src/mtev_http.c

```diff
@@ -1184,6 +1226,162 @@ mtev_http_session_req_consume(mtev_http_session_ctx *ctx,
   return bytes_read;
 }
 
+/* this magic GUID is defined in the websocket specification and must
+ * be what is used to create the accept key
+ * 
+ * See: https://tools.ietf.org/html/rfc6455#page-6
+ */
+#define WS_ACCEPT_KEY_GUID "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"
+
+void
+mtev_http_create_websocket_accept_key(char *dest, size_t dest_len, const char *client_key)
+{
+  SHA_CTX ctx;
+  unsigned char sha1[SHA_DIGEST_LENGTH], key_src[60];
+
+  memcpy(key_src, client_key, 24);
+  memcpy(key_src+24, WS_ACCEPT_KEY_GUID, 36);
```

Comment:
can these magic constants (`24`, `36`) be `#define`s? The `36` should presumably be `UUID_STR_LEN`, not off-hand sure where `24` comes from...